.changeset/fresh-experts-mix.md

```diff
@@ -0,0 +1,6 @@
+---
+"@firebase/database-compat": patch
+"@firebase/database": patch
+---
+
+Fixed issue where hostname set by `useEmulator` was being overridden by longpolling response
```

Comment:
It seems like the fix here is addressing an issue with the handshake overriding the host. IIUC that's not specific to longpolling.

Reply:
I mention longpolling specifically because we don't seem to change hosts for webosckets when using the emulator.